pysquared/state_of_health.py

```diff
@@ -21,154 +20,64 @@ class StateOfHealth:
+        ] = args
 
-    def system_voltage(self) -> float | None:
+    def get(self) -> OrderedDict[str, object]:
         """
-        Get the system voltage
-
-        :return: The system voltage in volts
-        :rtype: float | None
-
+        Get the state of health
         """
-        if self.battery_power_monitor is not None:
-            voltage: float = 0
-            try:
-                for _ in range(50):
-                    voltage += (
-                        self.battery_power_monitor.get_bus_voltage()
-                        + self.battery_power_monitor.get_shunt_voltage()
-                    )
-                return voltage / 50
-            except Exception as e:
-                self.logger.error("Couldn't acquire system voltage", err=e)
-
-    def battery_voltage(self) -> float | None:
+        state: OrderedDict[str, object] = OrderedDict()
+
+        for sensor in self._sensors:
+            if isinstance(sensor, Processor):
+                name = sensor.__class__.__name__
+                state[f"{name}_temperature"] = sensor.temperature
+            if isinstance(sensor, Flag):
+                state[sensor.get_name()] = sensor.get()
+            if isinstance(sensor, Counter):
+                state[sensor.get_name()] = sensor.get()
+            if isinstance(sensor, RadioProto):
+                name = sensor.__class__.__name__
+                state[f"{name}_modulation"] = sensor.get_modulation().__name__
+            if isinstance(sensor, PowerMonitorProto):
+                name = sensor.__class__.__name__
+                state[f"{name}_current_avg"] = self.avg_readings(sensor.get_current)
+                state[f"{name}_bus_voltage_avg"] = self.avg_readings(
+                    sensor.get_bus_voltage
+                )
+                state[f"{name}_shunt_voltage_avg"] = self.avg_readings(
+                    sensor.get_shunt_voltage
+                )
+            if isinstance(sensor, TemperatureSensorProto):
+                name = sensor.__class__.__name__
+                state[f"{name}_temperature"] = sensor.get_temperature()
+
+        self._log.info("State of Health", state=state)
+
+        return state
+
+    def avg_readings(
+        self, func: Callable[..., float | None], num_readings: int = 50
+    ) -> float | None:
         """
-        Get the battery voltage
+        Get the average of the readings from a function
 
-        :return: The battery voltage in volts
+        :param func: The function to call
+        :param num_readings: The number of readings to take
+        :return: The average of the readings
         :rtype: float | None
-
-        """
-        if self.battery_power_monitor is not None:
-            voltage: float = 0
-            try:
-                for _ in range(50):
-                    voltage += self.battery_power_monitor.get_bus_voltage()
-                return voltage / 50 + 0.2  # volts and correction factor
```

Comment:
This correction factor is no longer being applied. We could talk about it though.